apps/mobile/src/contexts/GalleryEditor/GalleryEditorContext.tsx

```diff
@@ -353,15 +377,48 @@ const GalleryEditorProvider = ({ children, queryRef }: Props) => {
           id: tokenId,
           kind: 'token',
           tokenRef,
+          communityName: tokenRef.definition.community?.name ?? '',
+          communityDescription: tokenRef.definition.community?.description ?? '',
+          communityId: tokenRef.definition.community?.dbid ?? '',
         });
       });
 
-      updateRow(activeRowId, (previousRow) => {
+      // Check if all the tokensToAdd have the same community
+      const communityIds = new Set(tokensToAdd.map((token) => token.communityId));
+
+      // if there are only tokens from the same community, then the section name should be the community name
+      if (communityIds.size === 1 && tokensToAdd[0]) {
+        sectionName = tokensToAdd[0].communityName;
+        sectionCollectorsNote = tokensToAdd[0].communityDescription;
+      }
+
+      let isNewSection = false;
+
+      updateRow(rowId, (previousRow) => {
+        if (previousRow.items.length === 0) {
+          isNewSection = true;
+          //  if the total number of tokens is less than 6, then the number of columns should be equal to the number of tokens
+          //  if more than 6, then the number of columns should be equal to 6
+          const columns = tokensToAdd.length < 6 ? tokensToAdd.length : 6;
+          return { ...previousRow, items: tokensToAdd, columns };
+        }
```

Comment:
If its a new section/row, the row will be updated based on how many tokens been added